src/subcommand/wallet/offer/create.rs

```diff
@@ -0,0 +1,98 @@
+use super::*;
+
+// TODO:
+// - [ ] add a --dry-run flag that doesn't sign our inputs in the PSBT
+// - [ ] pub fee: u64,
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Output {
+  pub psbt: String,
+  pub seller_address: Address<NetworkUnchecked>,
+  pub inscription: InscriptionId,
+}
+
+#[derive(Debug, Parser)]
+pub(crate) struct Create {
+  #[arg(long, help = "<INSCRIPTION> to make offer for.")]
+  inscription: InscriptionId,
+  #[arg(long, help = "<AMOUNT> to offer.")]
+  amount: Amount,
+  #[arg(long, help = "<FEE_RATE> for finalized transaction.")]
+  fee_rate: FeeRate,
+}
+
+impl Create {
+  pub(crate) fn run(&self, wallet: Wallet) -> SubcommandResult {
+    ensure!(
+      !wallet.inscription_info().contains_key(&self.inscription),
+      "inscription {} already in our wallet",
+      self.inscription
+    );
+
+    let inscription = wallet.get_inscription(self.inscription)?;
+
+    let Some(seller_address) = inscription.address else {
+      bail!(
+        "inscription {} script pubkey not valid address",
+        self.inscription,
+      );
+    };
+
+    let seller_address = seller_address
+      .parse::<Address<NetworkUnchecked>>()
+      .unwrap()
+      .require_network(wallet.chain().network())?;
+
+    let Some(postage) = inscription.value else {
+      bail!("inscription {} unbound", self.inscription);
+    };
+
+    let postage = Amount::from_sat(postage);
+
+    let tx = Transaction {
+      version: Version(2),
+      lock_time: LockTime::ZERO,
+      input: vec![TxIn {
+        previous_output: inscription.satpoint.outpoint,
+        script_sig: ScriptBuf::new(),
+        sequence: Sequence::ENABLE_RBF_NO_LOCKTIME,
+        witness: Witness::new(),
+      }],
+      output: vec![
+        TxOut {
+          value: postage,
+          script_pubkey: wallet.get_change_address()?.into(),
+        },
+        TxOut {
+          value: self.amount + postage,
+          script_pubkey: seller_address.clone().into(),
+        },
+      ],
+    };
+
+    let tx = fund_raw_transaction(wallet.bitcoin_client(), self.fee_rate, &tx)?;
```

Comment:
Am I tripping or is this is not protecting inscribed/runic UTXOs?

Reply:
your not tripping, good catch. Fixed